### PR TITLE
Replace scp with rsync

### PR DESCRIPTION
**Pull request by Annu149**
Security vulnerability was found in the scp program shipped with the openssh-clients package and CVSS score for this security vulnerability is 7.8 (https://access.redhat.com/security/cve/cve-2020-15778). Recommended action was to replace scp with rsync so,

 * Replaced scp with rsync in gpdb cm utilities and utility files
 * Renamed gpscp utility to gpsync after replacing scp with rsync in utility code

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: gpMgmt/bin/gpssh-exkeys [hunk 1]
```diff
@@ -755,7 +755,7 @@ try:
 
 
         for h in GV.newHosts:
-            cmd = ('scp -q -o "BatchMode yes" -o "NumberOfPasswordPrompts 0" ' +
+            cmd = ('rsync -q -e "ssh -o BatchMode=yes -o NumberOfPasswordPrompts=0" ' +
```
Comment (Sasasu) [inline gpMgmt/bin/gpssh-exkeys:758]: just a question. OpenSSH has a utility `/usr/bin/ssh-copy-id`

ssh-copy-id will handle file permission correctly (0o600 not 0o666), looks like we have a permission error for `authorized_keys` and we will overwrite the user config.

File: gpMgmt/bin/Makefile [hunk 2]
```diff
@@ -199,6 +199,6 @@ clean distclean:
 	rm -f analyzedbc gpactivatestandbyc gpaddmirrorsc gpcheckcatc \
 		  gpcheckperfc gpcheckresgroupimplc gpchecksubnetcfgc gpconfigc \
 		  gpdeletesystemc gpexpandc gpinitstandbyc gplogfilterc gpmovemirrorsc \
-		  gppkgc gprecoversegc gpreloadc gpscpc gpsdc gpssh-exkeysc gpsshc \
+		  gppkgc gprecoversegc gpreloadc gpsyncc gpsdc gpssh-exkeysc gpsshc \
```
Comment (piyushc01) [inline gpMgmt/bin/Makefile:202]: previous residual gpscc file may still be there. You can keep gpscpc also in the list for distclean

File: gpMgmt/bin/gppylib/commands/unix.py [hunk 3]
```diff
@@ -510,10 +510,10 @@ def canonicalize(addr):
...
 
-class Scp(Command):
+class Rsync(Command):
     def __init__(self, name, srcFile, dstFile, srcHost=None, dstHost=None, recursive=False, ctxt=LOCAL,
                  remoteHost=None):
-        cmdStr = findCmdInPath('scp') + " "
+        cmdStr = findCmdInPath('rsync') + " "
 
         if recursive:
             cmdStr = cmdStr + "-r "
```
Comment (piyushc01) [inline gpMgmt/bin/gppylib/commands/unix.py:519]: Will rsync retain the file permissions?

File: gpMgmt/bin/gppylib/operations/buildMirrorSegments.py [hunk 4]
```diff
@@ -19,7 +19,7 @@
 from gppylib.operations.utils import ParallelOperation, RemoteOperation
 from gppylib.system import configurationInterface as configInterface
 from gppylib.commands.gp import is_pid_postmaster, get_pid_from_remotehost
-from gppylib.commands.unix import check_pid_on_remotehost, Scp
+from gppylib.commands.unix import check_pid_on_remotehost, Rsync
```
Comment (piyushc01) [inline gpMgmt/bin/gppylib/operations/buildMirrorSegments.py:22]: Is rsync function used anywhere in this file?
I think if not used, can remove it from here.

File: gpMgmt/bin/gppylib/operations/test/regress/test_package/__init__.py [hunk 5]
```diff
@@ -10,11 +10,11 @@
 from gppylib.gpversion import MAIN_VERSION
 from contextlib import closing
 from gppylib.commands import gp
-from gppylib.commands.unix import Scp
+from gppylib.commands.unix import Rsync
 from gppylib.commands.base import Command, ExecutionError, REMOTE
 from gppylib.operations import Operation
 from gppylib.operations.unix import CheckFile, CheckRemoteFile, RemoveRemoteFile
-from gppylib.operations.package import dereference_symlink, GpScp, linux_distribution_id, linux_distribution_version
+from gppylib.operations.package import dereference_symlink, GpRsync, linux_distribution_id, linux_distribution_version
```
Comment (piyushc01) [inline gpMgmt/bin/gppylib/operations/test/regress/test_package/__init__.py:17]: If class Gprsync is not being used, can be removed safely from the import list

File: gpMgmt/bin/gpssh-exkeys (hunk 1 above)
Comment (Annu149) [reply gpMgmt/bin/gpssh-exkeys:754]: Thanks for pointing it out. For now we are just replacing scp with rsync in the codebase and haven't faced any permission issues till now. Still I will open a new jira to do an analysis on ssh-copy-id and try to address above concern.

**Review by ravoorsh — APPROVED** [review]
Looks good to me.

**Pull request merged**